### PR TITLE
Add PostgreSQL service to CI environment

### LINKED ISSUE #47: DB tests cannot run in CI environment.
This is because the PostgreSQL container has yet to be set up.

### PR DESCRIPTION
**Pull request by RichDom2185**
Loosely followed guide from https://docs.github.com/en/actions/using-containerized-services/creating-postgresql-service-containers.

This allows DB-related tests to be run in CI.

The corresponding tests have also been un-skipped so that they will run in the CI environment.

Closes #47.

### CODE REVIEW

**Comment by RichDom2185** [comment]
@YaleChen299 this is also blocked by #39 as we need to first create the DB before `connect_test` can function.

**Comment by RichDom2185** [comment]
> @YaleChen299 this is also blocked by #39 as we need to first create the DB before `connect_test` can function.

I decided we can use the `postgres` default DB for now, but once #39 is merged, then we will use our own DB.

**Review by ngruiyanrena — APPROVED** [review]

**Pull request merged**
